metrics/metrics.go

```diff
@@ -27,8 +27,8 @@ const (
 
 var (
 	// nodePodsCount tracks how many pods are nodes by type and by node name.
-	nodePodsCount = prometheus.NewCounterVec(
-		prometheus.CounterOpts{
+	nodePodsCount = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
```

Comment:
The change here is because in the code handling updates of values for this metric, they were calling Set(), and this isn't something you should do for Counters at all. It was discouraged in the version of prometheus used originally, but literally removed in this newer version. Using a Gauge when you need to hard set specific values rather than increment or decrement is the documented way to go.